modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexTaskAction.java

```diff
@@ -93,52 +90,43 @@ public boolean getWaitForCompletion() {
 
     public static class Response extends ActionResponse {
 
-        static final ParseField TASK_ID = new ParseField("task_id");
-        static final ParseField REINDEX_RESPONSE = new ParseField("reindex_response");
-
-        private static final ConstructingObjectParser<Response, Void> PARSER = new ConstructingObjectParser<>(
-            "start_reindex_response", true, args -> new Response((String) args[0], (BulkByScrollResponse) args[1]));
-
-        static {
-            PARSER.declareString(ConstructingObjectParser.constructorArg(), TASK_ID);
-            PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(),
-                (parser, context) -> BulkByScrollResponse.fromXContent(parser), REINDEX_RESPONSE);
-        }
-
         private final String taskId;
+        private final String ephemeralTaskId;
         @Nullable private final BulkByScrollResponse reindexResponse;
 
-        public Response(String taskId) {
-            this(taskId, null);
+        public Response(String taskId, String ephemeralTaskId) {
+            this(taskId, ephemeralTaskId, null);
         }
 
-        public Response(String taskId, BulkByScrollResponse reindexResponse) {
+        public Response(String taskId, String ephemeralTaskId, BulkByScrollResponse reindexResponse) {
```

Comment:
AFAICS, this means we are now returning the persistent task id out of the `RestReindexAction`. I wonder if we should explicitly name the persistent variant here `persistentTaskId` and then also add that to the rest-response?
I think we need to keep the rest response's `task` field to be the ephemeral task id in order to be BWC. It might therefore be beneficial to be explicit with the naming, that is using `persistent_task` in the rest response?
Alternatively we could be more subtle and stay BWC with the `task` field, but add `task_id` being the new persistent variant. This would primarily be a 7.x issue and thus might be a worthwhile compromise?